src/runtime_src/driver/xclng/drm/xocl/subdev/icap.c

```diff
@@ -152,6 +152,10 @@ struct icap {
 	char			*bit_buffer;
 	unsigned long		bit_length;
 	char			*icap_clock_freq_counter_hbm;
+
+	struct xcl_hwicap	cache;
+	ktime_t			cache_expires;
```

Comment:
not sure if we can cache icap data. e.g., after user programed clock, they should see the real counter value immediately, right?